### PR TITLE
Add tests for header service

### PR DESCRIPTION
**Pull request by pawellewandowski98**
Description:
---
- Add tests for header service

### CODE REVIEW

File: service/header_service_test.go [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+		Peers:        nil,
+	})
+
+	return &testData{
+		db: &array,
+		hs: hs,
+	}
+}
+
+func createFifthHeader() domains.BlockHeader {
```
Comment (wregulski) [inline service/header_service_test.go:272]: Maybe let's merge those two funcs as they are identical, and pass to them correct parameters creating fork and fifth header and then name the variables as:
fork,
fifthHeader

**Review by wregulski — CHANGES_REQUESTED** [review]

File: service/chain_service_test.go [hunk 2]
```diff
@@ -280,12 +280,14 @@ func assertHeaderInState(t *testing.T, h *domains.BlockHeader, s domains.HeaderS
...
 	db, tip := testrepository.StartingChain()
-	return testrepository.NewTestRepositories(db), tip
+	return testrepository.NewTestRepositories(&db), tip
 }
 
 func givenLongestChainInRepository() (repository.Repositories, *domains.BlockHeader) {
 	db, tip := testrepository.LongestChain()
-	return testrepository.NewTestRepositories(db), tip
+
+	var array []domains.BlockHeader = db
```
[the remark this answers is not in the document]
Comment (pawellewandowski98) [reply service/chain_service_test.go:289]: It looks weird but after adding pointer to `db *[]domains.BlockHeader` in `header_testrepository` I can not create repository directly from variable with type `testrepository.headerChainFixture`.

File: service/header_service_test.go [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+}
+
+var blocksToConfirmFork = 4
+
+func TestAddHeader(t *testing.T) {
+	tData := setUpServices()
+	tData.hs.Headers.AddHeader(createFifthHeader(), blocksToConfirmFork)
+
+	assert.Equal(t, len(*tData.db), 6)
+}
```
[the remark this answers is not in the document]
Comment (pawellewandowski98) [reply service/header_service_test.go:24]: Ok, I removed this method

File: internal/tests/testrepository/header_testrepository.go [hunk 4]
```diff
@@ -153,40 +150,73 @@ func (r *HeaderTestRepository) GenesisExists() bool {
...
+	}
+
+	return tips, nil
 }
 
-// GetChainBetweenTwoHashes empty test.
+// GetChainBetweenTwoHashes returns all headers between two hashes.
 func (r *HeaderTestRepository) GetChainBetweenTwoHashes(low string, high string) ([]*domains.BlockHeader, error) {
-	return nil, nil
+	h1 := findHeader(low, *r.db)
```
[the remark this answers is not in the document]
Comment (pawellewandowski98) [reply internal/tests/testrepository/header_testrepository.go:191]: Changed to hLow and hHigh

File: internal/tests/testrepository/test_repository.go [hunk 5]
```diff
@@ -6,7 +6,7 @@ import (
 )
 
 // NewTestRepositories creates repository.Repositories for unit testing usage.
-func NewTestRepositories(db []domains.BlockHeader) repository.Repositories {
+func NewTestRepositories(db *[]domains.BlockHeader) repository.Repositories {
 	return repository.Repositories{
 		Headers: NewHeadersTestRepository(db),
 	}
```
Comment (dorzepowski) [inline internal/tests/testrepository/test_repository.go:12]:
```suggestion
func NewTestRepositories(db []domains.BlockHeader) repository.Repositories {
	return repository.Repositories{
		Headers: NewHeadersTestRepository(&db),
	}
```

File: service/chain_service_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (dorzepowski) [reply service/chain_service_test.go:289]: I've checked that and IMHO to make the usage easier lets remove `*` from [argument to NewTestRepositories](https://github.com/libsv/pulse/pull/89/files#diff-7634bde107366d3e43caa883e2e401717712eb33efb701337bc86e0b8295754bR9) and add `&` in [line with creation of test header repository](https://github.com/libsv/pulse/pull/89/files#diff-7634bde107366d3e43caa883e2e401717712eb33efb701337bc86e0b8295754bR11)

In other words make the change like in https://github.com/libsv/pulse/pull/89/files#r1158536841

**Review by sirdeggen — APPROVED** [review]

**Review by wregulski — APPROVED** [review]
